Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/ThresholdedRandomCutForest.java

```diff
@@ -170,14 +171,31 @@ public ThresholdedRandomCutForest(ForestMode forestMode, TransformMethod transfo
         this.lastAnomalyDescriptor = descriptor;
     }
 
-    // for conversion from other thresholding models
-    public ThresholdedRandomCutForest(RandomCutForest forest, double futureAnomalyRate, List<Double> values) {
+    // this constructor produces an internally shingled ThresholdedRCF model from an
+    // externally shingled ThresholdedRCF model (which may or may not be in use in
+    // older
+    // versions of OpenSearch) absent any transformations and augmentations.
+    // A benefit of this conversion would be that imputations would be accessible
+    // to ThresholdedRCF -- that is, even if not every value of the input tuple is
+    // known
+    // the function process() would be able to provide an anomaly score (which is
+    // likely near
+    // minimum, since RCF is used to fill in the missing values). As a result, high
+    // values of the
+    // anomaly score will continue to be likely anomalies.
+    // Note that the basic RandomCutForest cannot be changed easily
+    // but the process() function would only require a fraction of the input
+    // see ThresholdedRandomCutForestMapperTest
+    public ThresholdedRandomCutForest(RandomCutForest forest, double futureAnomalyRate, List<Double> values,
+            double[] lastShingledInput) {
```

Comment:
Null/double[0] will likely cause exception -- the length has to be inputDimension*shingleSize (say 2 and 5). Then using new double [10] (all 0 values) or just pretending that the last known values are repeated shingleSize times are probably the best available options (need not be good options! -- if resources permit, it may be better to rebuild models, but if that is not an option the 0's or last known values are the ways to go).